### DISCUSSION TITLE
There is an error for a special PDF file.

### DISCUSSION THREAD

**Discussion by ling3882688** [sumatrapdfreader/sumatrapdf#4738] [Q&A]
[47.pdf](https://github.com/user-attachments/files/18261210/47.pdf)
![微信图片_20241227220051](https://github.com/user-attachments/assets/849426ce-ab7e-47c2-87e2-28fc2fe4226c)

I can open this PDF and print it manually,but an error occurred with command line.
The same command line for any other PDF files is ok,only this file has an error.

**Comment by GitHubRulesOK** [comment]
command lines need to work within the local constraints where the printer name is as seen by the system language.
I dont read Chinese to know those parts of your instruction but try `-print-to "printer name" ..... "c:\Users........pdf"` 
You may need the console to be in international  `chcp 65001` mode for UTF handover

**Reply by ling3882688** [reply]
The same command line for any  other PDF files is ok,only this file has an error.
I have uploaded the pdf file.U can test。

**Comment by GitHubRulesOK** [comment]
@ling3882688 
my suspicion is that if just "one file" is odd unless the filename characters are outside of normal command line range but the message suggests it is the printer name ? so something is odd.


For the benefit of others visiting this topic

For a batch file to switch console code pages the normal method is set before and reset after
```
REM important to allow for UTF input we switch cp to 65001
For /f "tokens=2 delims=:" %%G in ('chcp') Do set _codepage=%%G
chcp 65001 > nul
```
and later after the functional command line 
```
:eof
REM clean-up
chcp %_codepage% >nul
```

**Comment by GitHubRulesOK** [comment]
@ling3882688 the upload is not visible so wonder if the file is problematic ?

**Reply by ling3882688** [reply]
Thanks for your reply. It doesn't work.
I dont know why it is not visible and can not be downloaded，Can I send it  to your email？
I am sure the file is ok. It can be opened and printed unless command line.

**Reply by ling3882688** [reply]
Now the pdf file can be downloaded.
https://github.com/user-attachments/files/18261210/47.pdf

**Comment by GitHubRulesOK** [comment]
Interesting there is no problem with viewing in other readers and editors 
and print a smaller file works to produce one 500 times bigger ! 
most basic command `SumatraPDF-prerel-32.exe -print-to "Microsoft Print to PDF"  47.pdf`
yet for this one SumatraPDF command line handing over to "Print as PDF" triggers Windows to reply 
![image](https://github.com/user-attachments/assets/74f1ab16-7117-4ae8-a8b7-4c7bc6014351)

On a second run it works !!! I get 2 pages as over 7MB file
and have heard similar somewhere else in a different forum where related to Office printing the run need to be doubled
I have to "guess" the amount of initial memory required is causing an issue. but why a second run passes is unknown

**Reply by ling3882688** [reply]
Actually the pdf contains 5 pages，I delete page 3.4.5 for test.
It's really strange.There must be some other files as same as this file.
I guess there is something wrong with sumatrapdf command line mode.I tried V3.5 and V3.6，same errors.

**Reply by GitHubRulesOK** [reply]
I suggest you try the 2 page version to see if a second run works for you 
I had not pushed to issues as it is best you raise one there but the file should be the 5 page version for developer to test the real functionality perhaps if too large to upload consider email to developer @kjk whose gmail is on his profile https://github.com/kjk

**Reply by ling3882688** [reply]
Truly appreciate.
I will contact him soon.

Second run is all the same.

**Comment by GitHubRulesOK** [comment]
The original file no longer is an issue printing in current Pre-Release the command line works without error when filename is acceptable within the current OS. So must have been a Windows fiilename issue at the time.
This works without throwing a cannot print.
<img width="639" height="222" alt="image" src="https://github.com/user-attachments/assets/9d870672-6cef-42fb-8dd1-5180655fd1a8" />
<img width="610" height="195" alt="image" src="https://github.com/user-attachments/assets/7687cb51-5319-4610-b3d8-b12d546bd32e" />

**Reply by ling3882688** [reply]
Ok,Thanks.
I'll try 3.7 pre-release.
